nav2_robot/src/robot.cpp

```diff
@@ -16,22 +16,75 @@
 
 #include <string>
 #include <exception>
+
 #include "urdf/model.h"
 
 namespace nav2_robot
 {
 
-Robot::Robot(rclcpp::Node::SharedPtr & node)
-: node_(node), initial_pose_received_(false), initial_odom_received_(false)
+Robot::Robot(nav2_lifecycle::LifecycleNode::SharedPtr node)
+: node_(node)
+{
+  RCLCPP_INFO(node_->get_logger(), "Robot: Creating");
+}
+
+Robot::~Robot()
+{
+  RCLCPP_INFO(node_->get_logger(), "Robot: Destroying");
+}
+
+nav2_lifecycle::CallbackReturn
+Robot::on_configure(const rclcpp_lifecycle::State & /*state*/)
 {
-  // TODO(mhpanah): Topic names for pose and odom should should be configured with parameters
+  RCLCPP_INFO(node_->get_logger(), "Robot: Configuring");
+
+  // This class may be used from a module that comes up after AMCL has output
+  // its initial pose, so that pose message uses durability TRANSIENT_LOCAL
+  rmw_qos_profile_t pose_qos_profile = rmw_qos_profile_default;
+  pose_qos_profile.durability = RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL;
+
   pose_sub_ = node_->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
-    "amcl_pose", std::bind(&Robot::onPoseReceived, this, std::placeholders::_1));
+    "amcl_pose", std::bind(&Robot::onPoseReceived, this, std::placeholders::_1), pose_qos_profile);
 
   odom_sub_ = node_->create_subscription<nav_msgs::msg::Odometry>(
-    "odom", std::bind(&Robot::onOdomReceived, this, std::placeholders::_1));
+    "odom", std::bind(&Robot::onOdomReceived, this, std::placeholders::_1),
+    rmw_qos_profile_sensor_data);
 
   vel_pub_ = node_->create_publisher<geometry_msgs::msg::Twist>("/cmd_vel", 1);
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+Robot::on_activate(const rclcpp_lifecycle::State & /*state*/)
```

Comment:
I noticed in nav2_utils and nav2_tasks, you passed in a state value, even though it isn't used. We should probably be consistent across packages. I think this way is better. I'll add a comment in the other PR too.